### PR TITLE
multiply misfit by 0.5 factor

### PR DESCRIPTION
**Pull request by dngoldberg**
Addresses [issue 17](https://github.com/EdiGlacUQ/fenics_ice/issues/17). 

Needs review @jrmaddison. This is surely not the most elegant way to do this, but I do not know how to integrate multiplicative constants into J.addto(); rather I had to handle each term in a different way.

```
        fac = 0.5
...
            u_PRP = InterpolationMatrix(
                P.T @ spdiags(fac * 1.0 / (u_std[obs_local] ** 2),
                              0, P.shape[0], P.shape[0]) @ P)

...
        J_term = space_new(J.space())
        InnerProductSolver(uf, l_u_obs, J_term, alpha=-2.0 * fac).solve()
        J_ls_term_u.addto(J_term)
...
        J_term = space_new(J.space())
        function_assign(J_term, fac * J_u_obs)
        J_ls_term_u.addto(J_term)
```

### CODE REVIEW

File: fenics_ice/solver.py [hunk 1]
```diff
@@ -1133,14 +1133,16 @@ def comp_J_inv(self, verbose=False):
         J = Functional(name="J")
 
         # The following evaluates
-        #   (P u - u_obs)^T R_u_obs^{-1} (P u - u_obs)
-        #   + (P v - v_obs)^T R_v_obs^{-1} (P v - v_obs)
+        #   0.5 * (P u - u_obs)^T R_u_obs^{-1} (P u - u_obs)
+        #   + 0.5 * (P v - v_obs)^T R_v_obs^{-1} (P v - v_obs)
```
Comment (jrmaddison) [inline fenics_ice/solver.py:1137]: Also need to update six comments below

File: fenics_ice/solver.py [hunk 2]
```diff
@@ -1150,10 +1152,10 @@ def comp_J_inv(self, verbose=False):
                 tolerance=0.0)
 
             u_PRP = InterpolationMatrix(
-                P.T @ spdiags(1.0 / (u_std[obs_local] ** 2),
+                P.T @ spdiags(fac * 1.0 / (u_std[obs_local] ** 2),
                               0, P.shape[0], P.shape[0]) @ P)
             v_PRP = InterpolationMatrix(
-                P.T @ spdiags(1.0 / (v_std[obs_local] ** 2),
+                P.T @ spdiags(fac * 1.0 / (v_std[obs_local] ** 2),
```
Comment (jrmaddison) [inline fenics_ice/solver.py:1158]: Can use `alpha=fac` in the first two terms below

**Review by jrmaddison — APPROVED** [review]

**Comment by dngoldberg** [comment]
hi @bearecinos @jrmaddison unfortunately this is in draft mode as i was unable to run the mpi version of the inv_sigma either in git actions or on bow (the only machine i currently have an installation). 

is it possible that the blocking issue is now addressed on bow? is there an alternative -- it would be good to merge this PR soon.

**Comment by bearecinos** [comment]
Hi @dngoldberg I think a good thing will be to just merge it an if something its broken try to fix/debugg the test after the merge ... will look at this as soon as I have some time after my oggm runs

**Comment by dngoldberg** [comment]
ok i will merge via clicking the "merge pull request" button!

**Pull request merged**